include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -234,6 +234,95 @@ def BundleImpl : FIRRTLImplType<"Bundle", [FieldIDTypeInterface]> {
   }];
 }
 
+def EnumImpl : FIRRTLImplType<"Enum", [FieldIDTypeInterface]> {
```

Comment:
I'm a little concerned about how the `FieldIDTypeInterface` works here, but it seems like it might be the way I would like to use it in something like expand whens.  Will probably have to prototype match exprs/stmts and start working on pass support to know for sure.

Reply:
I didn't originally have it but it seems necessary.  Width and probably reset inference need to be walking enums.